app/javascript/admin/genericAdminModal.js

```diff
@@ -0,0 +1,39 @@
+const genericAdminModal = (
```

Comment:
```suggestion
const adminModal = (
```
I think `generic` is a bit redundant in the naming here. Thinking about how we name other reusable components across our codebase (like crayons), and we opt to just call them `Modal` for example.